models/mixpanel__daily_events.sql

```diff
@@ -127,18 +118,14 @@ final as (
         number_of_users - number_of_new_users - number_of_repeat_users as number_of_return_users,
         trailing_users_28d,
         trailing_users_7d,
-        event_type || '-' || date_day as unique_key
+        {{ dbt_utils.generate_surrogate_key(['event_type', 'date_day']) }} as unique_key,
+        {{ mixpanel.date_today('dbt_run_date')}}
 
     from agg_event_days
 
     {% if is_incremental() %}
-
-    -- only return the most recent day of data
-    where date_day >= coalesce( (select max(date_day)  from {{ this }} ), '2010-01-01')
-
+    where date_day >= {{ mixpanel.mixpanel_lookback(from_date="max(date_day)", interval=var('lookback_window', 7), datepart='day') }}
```

Comment:
This model actually creates records about ~7 days into the future (see the screenshot how the starting date_day is 02/27 and I ran the model on 02/21). Therefore the lookback window of 7 is only giving us one day to lookback. I was able to replicate this being an issue by inserting a late arriving record that came in on 02/19 and it didn't capture the events on the 19th. It brought the event type in, but didn't capture the events. However, when I changed this to 14 days it was properly brought in.

I would recommend we either remove the lookback and keep the 27 days consistent that is already used above, or expand this to 14 days to properly look back 7 days into the past.

![image](https://github.com/fivetran/dbt_mixpanel/assets/74217849/2a8e27a1-4bf9-4741-aef8-ce0fc053ae66)

Reply:
Ah yes thank you. I added a `+ 7` to handle!